### PR TITLE
Add FAQ

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## Frequently Asked Questions
+
+### [Could you combine an ELI5 (explain it like I'm five) and a sales pitch to try and convince me?](https://www.reddit.com/r/softwarearchitecture/comments/isw3so/a_programming_language_to_make_concurrent/g5cg3aq/)
+
+Here we go -- Computers have increasingly more CPUs/cores, as opposed to more powerful ones. Multithreaded code is hard to reason about and error prone, but it is also necessary today to take advantage of all the computing resources in a machine. The goal of Alan is to be similar to multithreaded Go or Java in performance, but similar to Python brevity and simplicity.
```
Comment (dfellis) [inline FAQ.md:5]: Nit: drop the "here we go." Leave that to [Super Mario](https://www.youtube.com/watch?v=7VvkXA6xpqI). :)

**Review by dfellis — COMMENTED** [review]
Also missing the FAQ section for this particular request that triggered the creation of an FAQ: https://discordapp.com/channels/682284785519165454/692533177587204117/755453224319713352

File: FAQ.md (hunk 1 above)
Comment (depombo) [reply FAQ.md:5]: hahaahah

File: FAQ.md [hunk 2]
```diff
@@ -54,6 +54,8 @@ Alan already encourages writing code that can be parallelized through arrays and
...
+### The "source installation" necessary to contribute to Alan requires Python, Node.js and Rust. Why are there so many dependencies?
 
-Yes, it's a limitation of a dependency to [nexe](https://github.com/nexe/nexe) and not something we are using within the project. The upcoming nexe 4.0 release will hopefully resolve this. That said our dependency on nexe is intended to be a temporary and would go away once we self-host or bootstrap an Alan compiler.
\ No newline at end of file
+We went with tools we were familiar with or believed would accelerate our ability to prove to ourselves that this could work, and that is reflected in the language implementation as it exists today, but we have always intended to rewrite the Alan compiler in Alan. Then only the runtime(s) would be in different languages.
+
+Currently, the compiler is written in Typescript, the main runtime in Rust and the secondary runtime in Javascript. Python is required due to a limitation of a dependency to [nexe](https://github.com/nexe/nexe) and not something we are using within the project. The upcoming nexe 4.0 release will hopefully resolve this. That said our dependency on nexe is intended to be a temporary.
```
Comment (dfellis) [inline FAQ.md:61]: Nit: `to be temporary` instead.

**Review by dfellis — APPROVED** [review]
LGTM, but the macos tests are failing because macos testing on Github Actions seems to be flaky in general. :/

Should we just disable all MacOS testing for now?

File: FAQ.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Still thinking about the best approach to take on the GPGPU front: do it "right" up-front but support the main backends relatively early, or prove it out with a single universal backend that might literally fudge numbers on you depending on your hardware. Also, not a priority until we've got the language really working, so post-v0.2, at least. If you have experience in this realm and want to contribute, please reach out!
+
+### Is there any interoperability?
+
+We are still scoping out [FFI support](https://github.com/alantech/alan/pull/60/files) for bindings that plays nice with the auto parallelization that happens in the VM. Alan (transpiles to Javascript)[https://docs.alan-lang.org/transpile_js.html] which still offers IO concurrency but without the parallelization.
```
Comment (LoPoHa) [inline FAQ.md:59]:
```suggestion
We are still scoping out [FFI support](https://github.com/alantech/alan/pull/60/files) for bindings that plays nice with the auto parallelization that happens in the VM. Alan [transpiles to Javascript](https://docs.alan-lang.org/transpile_js.html) which still offers IO concurrency but without the parallelization.
```

Just a little mismatch.

**Review by LoPoHa — COMMENTED** [review]
One of the links doesn't render, because [ and ( where switched.

File: FAQ.md (hunk 3 above)
Comment (depombo) [reply FAQ.md]: Thanks :)

**Comment by depombo** [comment]
> LGTM, but the macos tests are failing because macos testing on Github Actions seems to be flaky in general. :/
> 
> Should we just disable all MacOS testing for now?

let's leave it for now, but we can disable it if the flakyness continues
